aodntools/timeseries_products/velocity_aggregated_timeseries.py

```diff
@@ -0,0 +1,316 @@
+            ## get and store deployment metadata
+            LATITUDE[index] = nc.LATITUDE.values
+            LONGITUDE[index] = nc.LONGITUDE.values
+            NOMINAL_DEPTH[index] = TStools.get_nominal_depth(nc)
+            source_file[index] = file
+            instrument_id[index] = get_instrumentID(nc)
+
+    ## add atributes
+    with open(TEMPLATE_JSON) as json_file:
+        attribute_dictionary = json.load(json_file)
+    variable_attribute_dictionary = attribute_dictionary['_variables']
+    global_attribute_dictionary = attribute_dictionary['_global']
+
+    ## set variable attrs
+    for var in list(ds.variables):
+        ds[var].setncatts(variable_attribute_dictionary[var])
+
+    ## set global attrs
+    timeformat = '%Y-%m-%dT%H:%M:%SZ'
+    file_timeformat = '%Y%m%d'
+
+    time_start = nc4.num2date(np.min(TIME[:]), time_units, time_calendar).strftime(timeformat)
+    time_end = nc4.num2date(np.max(TIME[:]), time_units, time_calendar).strftime(timeformat)
+    time_start_filename = nc4.num2date(np.min(TIME[:]), time_units, time_calendar).strftime(file_timeformat)
+    time_end_filename = nc4.num2date(np.max(TIME[:]), time_units, time_calendar).strftime(file_timeformat)
+
+    contributor_name, contributor_email, contributor_role = TStools.get_contributors(files_to_agg)
+    add_attribute = {
+                    'title':                    ("Long Timeseries Velocity Aggregated product: " + ', '.join(varlist) + " at " +
+                                                  site_code + " between " + time_start + " and " + time_end),
+                    'site_code':                site_code,
+                    'time_coverage_start':      time_start,
+                    'time_coverage_end':        time_end,
+                    'geospatial_vertical_min':  np.min(ds['DEPTH']),
+                    'geospatial_vertical_max':  np.max(ds['DEPTH']),
+                    'geospatial_lat_min':       np.min(ds['LATITUDE']),
+                    'geospatial_lat_max':       np.max(ds['LATITUDE']),
+                    'geospatial_lon_min':       np.min(ds['LONGITUDE']),
+                    'geospatial_lon_max':       np.max(ds['LONGITUDE']),
+                    'date_created':             datetime.utcnow().strftime(timeformat),
+                    'history':                  datetime.utcnow().strftime(timeformat) + ': Aggregated file created.',
+                    'keywords':                 ', '.join(varlist + ['AGGREGATED']),
+                    'rejected_files':           "\n".join(rejected_files),
+                    'contributor_name':        "; ".join(contributor_name),
+                    'contributor_email':       "; ".join(contributor_email),
+                    'contributor_role':        "; ".join(contributor_role)}
+
+    global_attribute_dictionary.update(add_attribute)
+    ds.setncatts(dict(sorted(global_attribute_dictionary.items())))
+
+    ds.close()
+
+
+    ## create the output file name and rename the tmp file
```

Comment:
another case for a function ``` output_name = rename_file(oldname,newname) ```.
Also, you need to check for overwrites (rename to an already existing one).

Personally, I don't think you need to use "tmp" files or to rename at all - just do the kind of checks you are doing here and above before processing/concat the files (reading attributes, etc). If you think on this steps (check stuff, create a valid name) as a validation step, then you only spend time in valid inputs.

For example, What is the use of an entire concatenated array in memory if you cant find the site_code, facility_code, contributors, etc? Check the easy things first...